var/spack/repos/builtin/packages/umt/package.py

```diff
@@ -0,0 +1,196 @@
+    variant("caliper", default=False, description="Enable Caliper performance timers")
+    variant("mpi", default=True, description="Enable MPI support (mandatory")
+    variant("mfem", default=False, description="Enable support for reading MFEM meshes")
+    variant("umpire", default=False, description="Enable use of Umpire memory library")
+
+    ###########################################################################
+    # package dependencies
+    ###########################################################################
+
+    #######################
+    # CMake
+    #######################
+    depends_on("cmake@3.13.3:", type="build")
+
+    #######################
+    # Dependencies
+    #######################
+    depends_on("mpi")
+    depends_on("conduit+fortran~hdf5")
+    depends_on("mfem", when="+mfem")
+    depends_on("caliper+fortran", when="+caliper")
+    depends_on("adiak", when="+caliper")
+    depends_on("umpire+fortran", when="+umpire")
+    depends_on("camp", when="+umpire")
+
+    ####################################################################
+    # Note: cmake, build, and install stages are handled by CMakePackage
+    ####################################################################
+
+    def _get_sys_type(self, spec):
+        sys_type = spec.architecture
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        return sys_type
+
+    @property
+    def cache_name(self):
+        hostname = socket.gethostname()
+        if "SYS_TYPE" in env:
+            hostname = hostname.rstrip("1234567890")
+        return "{0}-{1}-{2}@{3}.cmake".format(
+            hostname,
+            self._get_sys_type(self.spec),
+            self.spec.compiler.name,
+            self.spec.compiler.version,
+        )
+
+    def cmake_args(self):
+        options = []
+        return options
+
+    def _get_host_config_path(self, spec):
+        sys_type = spec.architecture
+        # if on llnl systems, we can use the SYS_TYPE
+        if "SYS_TYPE" in env:
+            sys_type = env["SYS_TYPE"]
+        host_config_path = "{0}-{1}-{2}-{3}.cmake".format(
+            socket.gethostname(), sys_type, spec.compiler, spec.dag_hash()
+        )
+        dest_dir = spec.prefix
+        host_config_path = os.path.abspath(join_path(dest_dir, host_config_path))
+
+        return host_config_path
+
+    # Override the parent class function until its fixed in spack.  (Greg Becker and Chris White are aware of bug)
+    def flag_handler(self, name, flags):
+        if name in ("cflags", "cxxflags", "cppflags", "fflags", "ldflags"):
+            return (None, None, None)  # handled in the cmake cache
+        return (flags, None, None)
+
+    def initconfig_compiler_entries(self):
+        spec = self.spec
+
+        if "umpire" in spec:
+            if "+rocm" in spec["umpire"]:
+                spec.compiler_flags["cxxflags"].append("-I{0}/include".format(spec["hip"].prefix))
```

Comment:
We should make a `flags_to_build_system_args` method for `CachedCMakePackage` so that we don't have to do things like this, but that might be future work